crypto/evp/m_sha3.c

```diff
@@ -416,13 +329,13 @@ EVP_MD_SHAKE(256)
 const EVP_MD *evp_keccak_kmac##bitlen(void)     \
 {                                               \
     static const EVP_MD kmac_##bitlen##_md = {  \
-        -1,                                     \
+        NID_kmac##bitlen,                       \
```

Comment:
This is terribly confusing, as we also have EVP_MAC algorithms using the same name.  Although, I guess that as long as the desired operation is known (MAC or DIGEST), it's actually OK if they share the same name.  Is that an OK way to look at it?

Reply:
I agree, that it is confusing - as its kind of a sub algorithm of another algorithm.